src/mds/Server.cc

```diff
@@ -5704,6 +5704,10 @@ void Server::handle_set_vxattr(MDRequestRef& mdr, CInode *cur)
     try {
       rank = boost::lexical_cast<mds_rank_t>(value);
       if (rank < 0) rank = MDS_RANK_NONE;
+      else if (rank >= mds->mdsmap->get_max_mds()) {
```

Comment:
We've gotten this request before. I'm not fond of it because max_mds may go up or down for various reasons (especially with dynamic max_mds changes in the future) and this may break scripts.

Your change made me think however that we should check:

```suggestion
      else if (rank >= MAX_MDS) {
```

as that **is** an error.

Reply:
done.